core/utils/utils.go

```diff
@@ -1068,3 +1069,23 @@ func TryParseHex(s string) (b []byte, err error) {
 	}
 	return
 }
+
+// MinKey returns the minimum value of the given element array with respect
+// to the given key function. In the event U is not a compound type (e.g a
+// struct) an identity function can be provided.
+func MinKey[U any, T constraints.Ordered](elems []U, key func(U) T) T {
```

Comment:
Sidenote: methods like this would be very useful to have in a general utility library (honestly, most of the functions in the `utils` package can probably be safely extracted)